config/routes.rb

```diff
@@ -146,7 +146,14 @@
       resources :participants, controller: "schools/participants", only: :index
     end
 
-    resources :participants, only: %i[show index]
+    resources :participants, only: %i[show index] do
+      member do
+        scope path: "validations", controller: "participants/validations" do
+          get ":step", action: :show, as: :validation_step
+          post ":step", action: :update
+        end
+      end
```

Comment:
would this be easy to make resourceful?

Reply:
Not 100% resourceful. The relation between create and update are a bit bleak here - you cannot really create or change validation_step - those are defined by hardcoded array depending on the type of profile.

Potentially, each step could be seen as a separate resource without a create action. However, I had a bit of a hard time imagining how that would translate to resource. Theoretically, each form submission is not an update of the record but rather a completely new decision (and we will later on display the history of decisions). For now, I've decided to use papertrail for that, but I'm really not happy about this. Actually, event_log would be much better match here....